lib/vocabularies/discriminator/index.ts

```diff
@@ -21,13 +21,17 @@ const def: CodeKeywordDefinition = {
   error,
   code(cxt: KeywordCxt) {
     const {gen, data, schema, parentSchema, it} = cxt
-    const {oneOf} = parentSchema
+
+    const allowedKeywords = ['oneOf', 'anyOf'];
+    const keyword = allowedKeywords.find(key => Object.keys(parentSchema).includes(key));
```

Comment:
This is not very effective and this is on the hot path so I would optimize it:

```suggestion
    const keyword = Object.hasOwnProperty('oneOf')
      ? 'oneOf'
      : Object.hasOwnProperty('anyOf')
      ? 'anyOf'
      : undefined;
```